lib/saluki-core/examples/basic_supervisor.rs

```diff
@@ -0,0 +1,175 @@
+use std::time::Duration;
```

Comment:
I like how the API works. Probably worth developing a lint or something to confirm that processes actually check their shutdown signal properly. Shame we don't have the ability to preempt.